File: 2.linked-list/single-linked-list/rotate-list/rotate.js
const rotateRight = function (head, k) {
  if (!head) return head;
  let size = 0;
  let current = head;
  while (current) {
    size++;
    current = current.next;
  }
  for (let i = 0; i < k; ++i) {
    fast = fast.next;
  }
  while (fast && fast.next) {
    fast = fast.next;
    slow = slow.next;
  }
  fast.next = head;
  const node = slow.next;
  slow.next = null;
  return node;
};

File: 2.linked-list/single-linked-list/rotate-list/rotate.rb
def rorate_right(head, k)
  return head if head.nil?
  size, current = 0, head
  while (current)
    size += 1
    current = current.next
  end
  k %= size
  slow = fast = head
  for _ in 0..(k - 1)
    fast = fast.next
  end
  while (fast and fast.next)
    slow = slow.next
    fast = fast.next
  end
  fast.next = head
  node = slow.next # link with fast
  slow.next = nil # Remove after slow
  return node
end
